core/src/main/java/org/web3j/tx/Contract.java

```diff
@@ -372,9 +372,11 @@ TransactionReceipt executeTransaction(
                     String.format(
                             "Transaction has failed with status: %s. "
                                     + "Gas used: %d. (not-enough gas?)",
-                            receipt.getStatus(), receipt.getGasUsed()));
+                            receipt.getStatus(),
+                            receipt.getGasUsedRaw() != null
+                                    ? receipt.getGasUsed()
+                                    : BigInteger.valueOf(0)));
```

Comment:
this should be implemented differently

Reply:
can you do something like "Gas used: unknown" if it is not available

Reply:
It would also be useful to log the revert reason here if it is available

Reply:
Updated this to return unknown gas used, showing a revert reason would introduce a cyclic dependency.